app/src/main/java/com/task/wifitask/Base/DatabaseWiFi.java

```
@@ -0,0 +1,185 @@
+package com.task.wifitask.Base;
+
+
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+
+import androidx.annotation.NonNull;
+
+import com.task.wifitask.Entity.WiFiInfo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DatabaseWiFi implements RepositoryManipulation<WiFiInfo>{
+
+    public static DatabaseWiFi INSTANSE;
```

почему так? это синглтон? если да, то он потокобезопасный? ленивая инициализация или fail fast?

---

Ленивая, обусловленно это тем, что AplicationContext() можно получить в потоке сервиса или в потоке MainActivity. Насколько я знаю, DatabaseWiFi потокобезопасен.